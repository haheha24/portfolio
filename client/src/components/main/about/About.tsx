import "./about.css";
import Skills from "./Skills";
import { useReducer } from "react";

export enum SkillsMatch {
  front = "front",
  back = "back",
  general = "general",
  interp = "interp",
}
interface ISelectedState {
  front: { isActive: boolean; active?: string; inactive?: string };
  back: { isActive: boolean; active?: string; inactive?: string };
  general: { isActive: boolean; active?: string; inactive?: string };
  interp: { isActive: boolean; active?: string; inactive?: string };
  skill: SkillsMatch;
}
interface ISelectedType {
  type: SkillsMatch;
}

const About = () => {
  //Tracks which descriptions to render
  const initialSelected: ISelectedState = {
    front: {
      isActive: true,
      active: "about-skills-btn about-skills-btn-Active",
      inactive: "about-skills-btn",
    },
    back: {
      isActive: false,
      active: "about-skills-btn about-skills-btn-Active",
      inactive: "about-skills-btn",
    },
    general: {
      isActive: false,
      active: "about-skills-btn about-skills-btn-Active",
      inactive: "about-skills-btn",
    },
    interp: {
      isActive: false,
      active: "about-skills-btn about-skills-btn-Active",
      inactive: "about-skills-btn",
    },
    skill: SkillsMatch.front,
  };

  const handleSelected = (
    state: ISelectedState,
    action: ISelectedType
  ): ISelectedState => {
    switch (action.type) {
      case "front":
        return {
          front: {
            isActive: state.front.isActive = true,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          back: {
            isActive: state.back.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          general: {
            isActive: state.general.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          interp: {
            isActive: state.interp.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          skill: action.type,
        };
      case "back":
        return {
          front: {
            isActive: state.front.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          back: {
            isActive: state.back.isActive = true,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          general: {
            isActive: state.general.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          interp: {
            isActive: state.interp.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          skill: action.type,
        };
      case "general":
        return {
          front: {
            isActive: state.front.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          back: {
            isActive: state.back.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          general: {
            isActive: state.general.isActive = true,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          interp: {
            isActive: state.interp.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          skill: action.type,
        };
      case "interp":
        return {
          front: {
            isActive: state.front.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          back: {
            isActive: state.back.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          general: {
            isActive: state.general.isActive = false,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          interp: {
            isActive: state.interp.isActive = true,
            active: state.front.active,
            inactive: state.front.inactive,
          },
          skill: action.type,
        };
      default:
        return state;
    }
  };

  const [selected, setSelected] = useReducer(handleSelected, initialSelected);

  const handleSkillBtn = (skill: SkillsMatch) => {
    setSelected({ type: skill });
  };

  return (
    <article className="text-container about-article">
      <p className="about-paragraph">
        Hello! My name is Adrian Cristallo. I am a web developer with a passion
        for making websites and coding. I am looking for opportunities of work in web
        development while in the process of studying a Bachelor of Computer Science. I have
        also completed a Bachelor of Arts, majoring in sociology and history. I
        am always ready to learn new skills as my goal is to
        make this path into a long-lasting career. On the side I enjoy a little
        bit of gardening, gaming, and landscaping. Creativity is my outlet and enjoyment
        in life.
      </p>
      <p className="about-paragraph">
        If you are interested or have any questions, I invite you to reach out to me either through email, GitHub or linkedIn!
      </p>
      <div className="about-skills-container">
        <div className="about-btn-container">
          <button
            type="button"
            className={
              selected.front.isActive
                ? selected.front.active
                : selected.front.inactive
            }
            onClick={() => handleSkillBtn(SkillsMatch.front)}
          >
            Frontend Skills
          </button>
          <button
            type="button"
            className={
              selected.back.isActive
                ? selected.back.active
                : selected.back.inactive
            }
            onClick={() => handleSkillBtn(SkillsMatch.back)}
          >
            Backend Skills
          </button>
          <button
            type="button"
            className={
              selected.general.isActive
                ? selected.general.active
                : selected.general.inactive
            }
            onClick={() => handleSkillBtn(SkillsMatch.general)}
          >
            General Dev Skills
          </button>
          <button
            type="button"
            className={
              selected.interp.isActive
                ? selected.interp.active
                : selected.interp.inactive
            }
            onClick={() => handleSkillBtn(SkillsMatch.interp)}
          >
            Interpersonal Skills
          </button>
        </div>
        <Skills skill={selected.skill} />
      </div>
    </article>
  );
};

export default About;
