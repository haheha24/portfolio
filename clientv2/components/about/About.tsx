"use client";

import Skills from "./Skills";
import React, { useState } from "react";
import {
  TextContainer,
  Paragraph,
  SkillsContainer,
  SkillsBtnContainer,
  SkillsBtn,
} from "./styles";

const About = () => {
  const [selected, setSelected] = useState("front");

  const handleSkillBtn = (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    setSelected(e.currentTarget.name);
  };

  return (
    <TextContainer>
      <Paragraph data-testid="introduction">
        Hello! My name is Adrian Cristallo. I am a web developer with a passion
        for making websites and coding. I am looking for opportunities of work in web
        development while in the process of studying a Bachelor of Computer Science. I have
        also completed a Bachelor of Arts, majoring in sociology and history. I
        am always ready to learn new skills as my goal is to
        make this path into a long-lasting career. On the side I enjoy a little
        bit of gardening, gaming, and landscaping. Creativity is my outlet and enjoyment
        in life.
      </Paragraph>
      <SkillsContainer>
        <SkillsBtnContainer data-testid="skills-btn-container">
          <SkillsBtn
            type="button"
            name="front"
            active={selected === "front" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
            data-testid="front"
          >
            Frontend Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="back"
            active={selected === "back" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
            data-testid="back"
          >
            Backend Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="general"
            active={selected === "general" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
            data-testid="general"
          >
            General Dev Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="inter"
            active={selected === "inter" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
            data-testid="inter"
          >
            Interpersonal Skills
          </SkillsBtn>
        </SkillsBtnContainer>
        <Skills skill={selected} />
      </SkillsContainer>
    </TextContainer>
  );
};

export default About;
