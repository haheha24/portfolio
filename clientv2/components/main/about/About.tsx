import Skills from "./Skills";
import React, { useState } from "react";
import {
  TextContainer,
  Paragraph,
  SkillsContainer,
  SkillsBtnContainer,
  SkillsBtn,
} from "./aboutStyledComponents";

const About = () => {
  const [selected, setSelected] = useState("front");

  const handleSkillBtn = (
    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
  ) => {
    setSelected(e.currentTarget.name);
  };

  return (
    <TextContainer>
      <Paragraph>
        Hello! My name is Adrian Cristallo. I am a web developer with a passion
        for making websites and coding. Looking for opportunities of work in web
        development while I study a Bachelor of Information Technology. I have
        also completed a Bachelor of Arts, majoring in sociology and history. I
        am always ready to learn new things and technologies as my goal is to
        make this path into a long-lasting career. On the side I enjoy a little
        bit of gardening and landscaping. Creativity is my outlet and enjoyment
        in life.
      </Paragraph>
      <SkillsContainer>
        <SkillsBtnContainer>
          <SkillsBtn
            type="button"
            name="front"
            active={selected === "front" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
          >
            Frontend Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="back"
            active={selected === "back" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
          >
            Backend Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="general"
            active={selected === "general" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
          >
            General Dev Skills
          </SkillsBtn>
          <SkillsBtn
            type="button"
            name="inter"
            active={selected === "inter" ? true : false}
            onClick={(e) => handleSkillBtn(e)}
          >
            Interpersonal Skills
          </SkillsBtn>
        </SkillsBtnContainer>
        <Skills skill={selected} />
      </SkillsContainer>
    </TextContainer>
  );
};

export default About;
