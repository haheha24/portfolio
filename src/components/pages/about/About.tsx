import React from 'react'

const About = () => {
    return (
        <div>
            Hi this is my about me page
        </div>
    )
}

export default About
