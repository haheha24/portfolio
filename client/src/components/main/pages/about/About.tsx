import "./about.css";

export type IhomeSubtitle = {
  itemId: string;
  itemName: string;
};

const About = () => {
  const introduction =
    "Hey! I'm Adrian Cristallo, a Front-End Developer with a passion for making websites and coding. Looking for opportunities of work in web development while I study a Bachelor of Information Technology :). I've also completed a Bachelor of Arts, majoring in sociology and history. So I guess you could say I really like learning! On the side I enjoy a little bit of gardening and landscaping. Creativity is my outlet and enjoyment in life.";
  return (
    <div className="about-grid">
      <div id="aboutLink" className="about-container">
        <section id="aboutMe" className="about-section">
          <h2 className="about-subtitle">About Me</h2>
          <p className="about-paragraph">{introduction}</p>
          <h3 className="about-subtitle">
            My technical skills I use for web development are:
          </h3>
          <p className="about-paragraph">
            <ul>
              <li>HTML</li>
              <li>CSS</li>
              <li>JavaScript</li>
              <li>TypeScript</li>
              <li>React</li>
              <li>Jest</li>
            </ul>
          </p>
          <h3 className="about-subtitle">
            {" "}
            Back-end technologies I have used in projects:
          </h3>
          <p className="about-paragraph">
            <ul>
              <li>Node.JS</li>
              <li>Express.JS</li>
              <li>Mongoose</li>
            </ul>
          </p>
          <h3 className="about-subtitle">
            Other technologies I have experience in using are:
          </h3>
          <p className="about-paragraph">
            <ul>
              <li>MongoDB Atlas Cloud</li>
              <li>
                NPM Packages and the various libraries and frameworks that are
                available
              </li>
              <li>
                GIT Version Control and Linux Commands using GitBash intergrated
                terminal
              </li>
            </ul>
          </p>
          <h3 className="about-subtitle">Interpersonal Skills and Values</h3>
          <p className="about-paragraph">
            My interpersonal skills are one of my most proudest achievements in
            life so far. I believe strongly that communication can resolve any
            issue, along with respect and understanding. These three skills and
            values have lead me to my successes in life and continue to define
            my decisions and actions. I courageously trust myself to achieve my
            goals in building a professional career in web development, and so
            should you believe in me when I say I can help you achieve your
            goals as well!
          </p>
        </section>
      </div>
    </div>
  );
};

export default About;
