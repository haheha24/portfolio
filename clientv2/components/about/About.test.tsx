import About from "./About";
import { cleanup, render, fireEvent } from "@testing-library/react";

afterEach(() => {
  cleanup();
});

describe("Tests the About component", () => {
  it("Contains the introduction", () => {
    const { container } = render(<About />);
    expect(container).toHaveTextContent(
      "Welcome! My name is Adrian Cristallo."
    );
  });
  it("Begins with Frontend Skills and then click Backend Skills", () => {
    const { getByRole } = render(<About />);
    expect(
      getByRole("button", { name: "Frontend Skills Button" })
    ).toHaveAttribute("data-active", "true");
    expect(
      getByRole("button", { name: "Backend Skills Button" })
    ).toHaveAttribute("data-active", "false");
    expect(getByRole("article", { name: "container" })).toHaveTextContent(
      "HTML"
    );

    fireEvent.click(getByRole("button", { name: "Backend Skills Button" }));

    expect(
      getByRole("button", { name: "Frontend Skills Button" })
    ).toHaveAttribute("data-active", "false");
    expect(
      getByRole("button", { name: "Backend Skills Button" })
    ).toHaveAttribute("data-active", "true");
    expect(getByRole("article", { name: "container" })).toHaveTextContent(
      "Node.JS"
    );
  });
});
