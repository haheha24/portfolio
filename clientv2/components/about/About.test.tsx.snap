// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests the About component renders the section component 1`] = `
.c5 {
  font-size: 1.125em;
  color: #fff;
  padding: 0;
  margin: 0;
  background-color: hsl(260,100%,33%);
  cursor: pointer;
}

.c5:hover {
  background-color: hsl(260,100%,60%);
}

.c9 {
  color: #fff;
  background-color: hsl(216,7%,25%);
}

.c0 {
  -webkit-scroll-margin-top: 75px;
  -moz-scroll-margin-top: 75px;
  -ms-scroll-margin-top: 75px;
  scroll-margin-top: 75px;
}

.c1 {
  margin: 0 20%;
}

.c2 {
  list-style-position: inside;
  line-height: 2em;
  text-align: justify;
}

.c11 {
  list-style-position: inside;
  line-height: 2em;
  text-align: justify;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-top: 1.25em;
  min-height: 380px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  font-size: 1.125em;
  background-color: hsl(260,100%,60%);
  padding: 6px;
  border-top-right-radius: 20px;
  border-top-left-radius: 20px;
  border: 2.5px solid hsl(260,100%,60%);
  cursor: pointer;
}

.c6:hover {
  background-color: hsl(260,100%,60%);
}

.c7 {
  font-size: 1.125em;
  background-color: hsl(260,100%,33%);
  padding: 6px;
  border-top-right-radius: 20px;
  border-top-left-radius: 20px;
  border: 2.5px solid hsl(260,100%,60%);
  cursor: pointer;
}

.c7:hover {
  background-color: hsl(260,100%,60%);
}

.c8 {
  padding: 20px;
  border-top-left-radius: 2.5px;
  border-top-right-radius: 2.5px;
  border-bottom-left-radius: 10px;
  border-bottom-right-radius: 10px;
  box-shadow: 10px 10px hsla(216,7%,14%,0.5);
  background-color: hsl(260,100%,60%);
}

.c10 {
  padding: 10px 0;
  background-color: initial;
}

@media only screen and (max-width:500px) {
  .c1 {
    margin: 0;
  }
}

@media only screen and (max-width:500px) {
  .c2 {
    text-align: left;
    width: 75%;
    margin: 0 auto;
  }
}

@media only screen and (max-width:500px) {
  .c4 {
    -webkit-flex-direction: column;
    -ms-flex-direction: column;
    flex-direction: column;
  }
}

@media only screen and (max-width:500px) {
  .c6 {
    width: 100%;
    border-radius: 0;
  }
}

@media only screen and (max-width:500px) {
  .c7 {
    width: 100%;
    border-radius: 0;
  }
}

@media only screen and (max-width:500px) {
  .c8 {
    border-radius: 0;
    box-shadow: 0 10px 10px hsla(216,7%,14%,0.5);
  }
}

@media only screen and (max-width:500px) {
  .c10 {
    text-align: left;
    width: 75%;
    margin: 0 auto;
  }
}

<article
  class="c0 c1"
>
  <p
    class="c2"
    data-testid="introduction"
  >
    Hello! My name is Adrian Cristallo. I am a web developer with a passion for making websites and coding. Looking for opportunities of work in web development while I study a Bachelor of Information Technology. I have also completed a Bachelor of Arts, majoring in sociology and history. I am always ready to learn new things and technologies as my goal is to make this path into a long-lasting career. On the side I enjoy a little bit of gardening and landscaping. Creativity is my outlet and enjoyment in life.
  </p>
  <div
    class="c3"
  >
    <div
      class="c4"
      data-testid="skills-btn-container"
    >
      <button
        class="c5 c6"
        data-testid="front"
        name="front"
        type="button"
      >
        Frontend Skills
      </button>
      <button
        class="c5 c7"
        data-testid="back"
        name="back"
        type="button"
      >
        Backend Skills
      </button>
      <button
        class="c5 c7"
        data-testid="general"
        name="general"
        type="button"
      >
        General Dev Skills
      </button>
      <button
        class="c5 c7"
        data-testid="inter"
        name="inter"
        type="button"
      >
        Interpersonal Skills
      </button>
    </div>
    <div
      class="c8"
      data-testid="Skills"
    >
      <h3
        class="c9 c10"
      >
        My technical skills I use for web development are:
      </h3>
      <ul
        class="c11"
      >
        <li>
          HTML
        </li>
        <li>
          CSS
        </li>
        <li>
          JavaScript
        </li>
        <li>
          TypeScript
        </li>
        <li>
          React
        </li>
        <li>
          Next.JS 
        </li>
        <li>
          Jest
        </li>
      </ul>
    </div>
  </div>
</article>
`;
